lxd/instance/instance_utils.go

```diff
@@ -126,7 +126,7 @@ func exclusiveConfigKeys(key1 string, key2 string, config map[string]string) (va
 }
 
 // ValidConfig validates an instance's config.
-func ValidConfig(sysOS *sys.OS, config map[string]string, profile bool, expanded bool) error {
+func ValidConfig(sysOS *sys.OS, config map[string]string, profile bool, expanded bool, instanceType instancetype.Type) error {
```

Comment:
@my5t3ry because you only provide `instanceType = instancetype.Any` when `profile = true`, then I think we can remove the `profile` argument from this function entirely, and just use `instanceType = instancetype.Any` to indicate profile validation (with an updated doc block comment explaining that). Otherwise we end up with two arguments that do very similar things which is confusing.